install/osx/scripts/postinstall

```diff
@@ -12,6 +12,36 @@ chown -R nobody:nobody "$RUN_DIR"
 
 PLIST=/Library/LaunchDaemons/org.foldingathome.fahclient.plist
 
+OS_MAJOR="`sw_vers -productVersion | cut -f1 -d.`"
+if [ "$OS_MAJOR" -ge 15 ]; then
+    # macOS 15+
+    username=_fahclient
+    realname="Foldingathome.org Client Service"
+    # try to create user if it does not exist
+    if ! id -u $username &>/dev/null; then
+        for (( uid = 499;; --uid )) ; do
```

Comment:
It's better to add a condition to end the loop. Otherwise, assuming all UIDs are in use, theoretically the loop could enter negative UIDs, which doesn't make sense. Something like `(( uid = 499; uid > 0; --uid ))`.

I don't know anything about the macOS-specific bits.